### ISSUE TITLE
npm install results in ERR! registry error parsing json

### ISSUE THREAD

**Issue by varunmehra03** [npm/npm#20798]
#### I'm opening this issue because:
Unable to install node modules 
ERR! registry error parsing json

#### What's going wrong?
The CDN returns this error when executing npm install <module>. 
Fastly error: unknown domain: registry.npmjs.org. Please check that this domain has been added to a service.
Although when I connect to VPN (US), I am able to install the packages.
Details: cache-sin18022-SIN

#### Verbose Log
npm verb lastModified Sun, 27 May 2018 22:36:03 GMT
npm http request GET http://registry.npmjs.org/yargs
npm http 500 http://registry.npmjs.org/yargs
npm verb bad json 
npm verb bad json <html>
npm verb bad json <head>
npm verb bad json <title>Fastly error: unknown domain registry.npmjs.org</title>
npm verb bad json </head>
npm verb bad json <body>
npm verb bad json <p>Fastly error: unknown domain: registry.npmjs.org. Please check that this domain has been added to a service.</p>
npm verb bad json <p>Details: cache-sin18028-SIN</p></body></html>
npm ERR! registry error parsing json
npm verb headers { server: 'Varnish',
npm verb headers   'retry-after': '0',
npm verb headers   'content-type': 'text/html',
npm verb headers   'cache-control': 'private, no-cache',
npm verb headers   connection: 'keep-alive, keep-alive',
npm verb headers   'x-served-by': 'cache-sin18028-SIN',
npm verb headers   'content-length': '257',
npm verb headers   'accept-ranges': 'bytes',
npm verb headers   date: 'Tue, 29 May 2018 10:20:13 GMT',
npm verb headers   via: '1.1 varnish' }
npm info retry will retry, error on last attempt: SyntaxError: Unexpected token <
npm info retry 

### supporting information:

 - `npm -v` prints: 2.15.0
 - `node -v` prints: 0.12.13
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?:
 - Network issues:
   - Geographic location where npm was run: India
   - If I access the npm registry via a VPN, it works fine
 - Environment:
   -  OS X.

**Comment by ghost** [comment]
Same

npm -v
3.10.10

node -v
v6.10.0

npm config get registry
http://registry.npmjs.org/

proxy and https-proxy set and have been working for months with no issue.

**Comment by BE55Roberts** [comment]
We're seeing this too. Any idea how to resolve it?

**Comment by burdiuz** [comment]
it seems I experience same issue with node v10.2.1, v8.11.2 and npm 5.6.0. I've tried to overwrite registry endpoint with ` --registry https://registry.npmjs.com` but some children dependencies still accessing `registry.npmjs.org` and installation fails. Probably have to find a way to redirect all requests to .com domain.

When updating npm
```
C:\>npm install npm -g
npm ERR! code E500
npm ERR! 500 Internal Server Error: https://registry.npmjs.org/npm/-/npm-6.1.0.tgz

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\burdi\AppData\Roaming\npm-cache\_logs\2018-05-29T16_28_38_766Z-debug.log
```
Log
```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   'npm',
1 verbose cli   '-g' ]
2 info using npm@5.6.0
3 info using node@v8.11.2
4 verbose npm-session a5647b195295ebea
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 304 https://registry.npmjs.com/npm 101664ms attempt #3 (from cache)
8 silly pacote tag manifest for npm@latest fetched in 101721ms
9 silly install loadIdealTree
10 silly install cloneCurrentTreeToIdealTree
11 silly install loadShrinkwrap
12 silly install loadAllDepsIntoIdealTree
13 silly resolveWithNewModule npm@6.1.0 checking installable status
14 silly pacote trying https://registry.npmjs.org/npm/-/npm-6.1.0.tgz by hash: sha512-e38cCtJ0lEjLXXpc4twEfj8Xw5hDLolc2Py87ueWnUhJfZ8GA/5RVIeD+XbSr1+aVRGsRsdtLdzUNO63PvQJ1w==
15 silly pacote data for sha512-e38cCtJ0lEjLXXpc4twEfj8Xw5hDLolc2Py87ueWnUhJfZ8GA/5RVIeD+XbSr1+aVRGsRsdtLdzUNO63PvQJ1w== not present. Using manifest.
16 http fetch GET 500 https://registry.npmjs.org/npm/-/npm-6.1.0.tgz 72550ms attempt #3
17 verbose stack Error: 500 Internal Server Error: https://registry.npmjs.org/npm/-/npm-6.1.0.tgz
17 verbose stack     at fetch.then.res (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\pacote\lib\fetchers\registry\fetch.js:42:19)
17 verbose stack     at tryCatcher (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\util.js:16:23)
17 verbose stack     at Promise._settlePromiseFromHandler (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\promise.js:512:31)
17 verbose stack     at Promise._settlePromise (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\promise.js:569:18)
17 verbose stack     at Promise._settlePromise0 (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\promise.js:614:10)
17 verbose stack     at Promise._settlePromises (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\promise.js:693:18)
17 verbose stack     at Async._drainQueue (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\async.js:133:16)
17 verbose stack     at Async._drainQueues (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\async.js:143:10)
17 verbose stack     at Immediate.Async.drainQueues (C:\Users\burdi\AppData\Roaming\nvm\v8.11.2\node_modules\npm\node_modules\bluebird\js\release\async.js:17:14)
17 verbose stack     at runCallback (timers.js:810:20)
17 verbose stack     at tryOnImmediate (timers.js:768:5)
17 verbose stack     at processImmediate [as _immediateCallback] (timers.js:745:5)
18 verbose cwd C:\
19 verbose Windows_NT 10.0.17134
20 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "npm" "-g"
21 verbose node v8.11.2
22 verbose npm  v5.6.0
23 error code E500
24 error 500 Internal Server Error: https://registry.npmjs.org/npm/-/npm-6.1.0.tgz
25 verbose exit [ 1, true ]
```

**Comment by chrissound** [comment]
```
npm ERR! registry error parsing json
npm ERR! Linux 4.14.11-coreos
npm ERR! argv "node" "/usr/local/bin/npm" "install" "--save-dev" "node-bourbon"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.3
```

This has taken down my ship. It hit an iceberg due to this error. May day!

**Comment by legodude17** [comment]
@chrissound Does this occur on the latest version of npm?

**Comment by burdiuz** [comment]
I figured out my problem, it appeared that I've had entry in my hosts file with old IP for registry.npmjs.org(I've used it with WSL over VPN). After removing it everything started to work.

**Comment by jamwjam** [comment]
> I figured out my problem, it appeared that I've had entry in my hosts file with old IP for registry.npmjs.org(I've used it with WSL over VPN). After removing it everything started to work.

This was the same case for me. I removed the old entry and it's fixed! Thanks @burdiuz !

**Comment by varunmehra03** [comment]
I had to do nothing from my end to get this fixed. It seems to be fixed now. 

Thanks!

**Comment by omrico** [comment]
> I figured out my problem, it appeared that I've had entry in my hosts file with old IP for registry.npmjs.org(I've used it with WSL over VPN). After removing it everything started to work.

Thanks @burdiuz !!
Same issue, just removed the record form /etc/hosts file (in my Ubuntu) and problem was solved!

**Comment by mkantautas** [comment]
````/etc/hosts```` file is cleared from the npm records, but when trying to access ````https://registry.npmjs.org/```` getting a 500 , then ````https://registry.npmjs.com/```` one works, but as @burdiuz  mentioned even with explicitly provided ````--registry https://registry.npmjs.com/```` some children dependencies are trying to access the ````.org```` domain

**Comment by mkantautas** [comment]
Resolved with ````nameserver 8.8.8.8```` to ````/etc/resolv.conf````
